rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -0,0 +1,186 @@
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_spaces_served_by_swh_use import (
+    get_spaces_served_by_swh_use,
+)
+from rct229.schema.config import ureg
+from rct229.schema.schema_enums import SchemaEnums
+from pint import Quantity
+from rct229.utils.assertions import assert_
+from rct229.utils.jsonpath_utils import find_exactly_one_with_field_value, find_all
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.utility_functions import find_exactly_one_schedule
+
+SERVICE_WATER_HEATING_USE_UNIT = SchemaEnums.schema_enums[
+    "ServiceWaterHeatingUseUnitOptions"
+]
+
+REQUIRED_USE_UNIT = [
+    SERVICE_WATER_HEATING_USE_UNIT.POWER,
+    SERVICE_WATER_HEATING_USE_UNIT.VOLUME,
+]
+
+
+def get_energy_required_to_heat_swh_use(
+    swh_use: dict, rmd: dict, building_segment: dict
+) -> dict[str, Quantity]:
+    """
+    This function calculates the total energy required to heat the SWH use over the course of a year in btu.  Note - this function does not work for service water heating uses with use_units == "OTHER".  In this case, it will return 0.
+
+    Parameters
+    ----------
+    swh_use: dict, service_water_heating_uses
+    rmd: dict, RMD at RuleSetModelDescription level
+    building_segment: dict, building_segment
+
+    Returns
+    ----------
+    energy_required_by_space: A dict where the keys are space_ids and values are the total energy required to heat the swh_use for that space.  If a swh_use is not assigned to any spaces, the key will be "NO_SPACES_ASSIGNED"
+
+    """
+
+    hourly_schedule_id = swh_use.get("use_multiplier_schedule")
+    hourly_schedule = (
+        find_exactly_one_schedule(rmd, hourly_schedule_id)
+        if hourly_schedule_id is not None
+        else None
+    )
+
+    distribution_system_id = swh_use.get("served_by_distribution_system", "")
+    distribution_system = find_exactly_one_with_field_value(
+        "$.service_water_heating_distribution_systems[*]",
+        "id",
+        distribution_system_id,
+        rmd,
+    )
+    supply_temperature = distribution_system.get("design_supply_temperature")
+    assert_(
+        supply_temperature is not None,
+        "A service water heating supply temperature must be provided",
+    )
+    swh_use_value = swh_use.get("use", 0)
+    inlet_temperature_schedule_id = distribution_system.get(
+        "entering_water_mains_temperature_schedule", ""
+    )
```

Comment:
Again, it is ok to not getting this value if conditions does not meet. Also, we should clarify what if the schedule is missing in this scenario. @KarenWGard

Reply:
missing raise undetermined outcome.